src/transport/raw/MessageHeader.cpp

```diff
@@ -131,6 +131,31 @@ uint16_t PayloadHeader::EncodeSizeBytes() const
     return static_cast<uint16_t>(size);
 }
 
+CHIP_ERROR PacketHeader::DecodeFixed(const System::PacketBufferHandle & buf)
+{
+    CHIP_ERROR err             = CHIP_NO_ERROR;
+    const uint8_t * const data = buf->Start();
+    uint16_t size              = buf->DataLength();
+    LittleEndian::Reader reader(data, size);
+    int version;
+
+    uint8_t msgFlags;
+    SuccessOrExit(err = reader.Read8(&msgFlags).StatusCode());
```

Comment:
This is duplicating a bunch of code from `Decode`.  Can that move to a helper method called from both, please?  Presumably taking a `LittleEndian::Reader &` as an argument.